astroid/nodes/node_ng.py

```diff
@@ -314,7 +313,7 @@ def statement(
         return self.parent.statement(future=future)
 
     def frame(
-        self,
+        self, *, future: Literal[None, True] = None
```

Comment:
Why not
```suggestion
        self, *, future: bool = False
```
?

Reply:
The idea behind it was to only allow two specific call options:
* `node.frame()`
* `node.frame(future=True)`

It doesn't really make sense to add `future=None` or even `future=False` if you don't support the future case.

--
After your comment I noticed that we probably need to add overloads here to make that happen.

```py
@overload
def frame(self) -> Union[<all types>]:
    ...
@overload
def frame(self, *, future: Literal[True]) -> Union[<all types>]:
    ...
```
This needs to be added to **all** frame definitions as overloads aren't inherited.

_The overloads for `statement()` also need to be fixed, but I can do that in a separate PR._

Reply:
This probably also need a `pylint: disable` comment, as there is a false-positive with the `arguments-differ` message and overloads: https://github.com/PyCQA/pylint/issues/5264

```py
    # pylint: disable-next=arguments-differ
    # https://github.com/PyCQA/pylint/issues/5264
```

Reply:
For reference, the PR to fix the `node.statement()` overload typing: #1308

Reply:
Turns out the overload idea doesn't actually work as I though. Sorry for the confusion. The default with `None` is needed for the parent calls.
```py
...
return self.parent.frame(future=future)
```
I've already opened another PR to fix the `node.statement` calls #1317.